data-serving/data-service/src/controllers/case.ts

```diff
@@ -61,113 +64,160 @@ export class CasesController {
+                            doc = await cursor.next();
+                        }
+                    } catch(err) {
+                        logger.error(`Error streaming case data: ${err}`);
+                        res.status(500).json('A server error occurred while streaming case data');
+                        return;
+                    }
+                    res.end();
+                
                     break;
 
                 case 'tsv':
                     res.setHeader('Content-Type', 'text/tsv');
-                    res.setHeader(
-                        'Content-Disposition',
-                        `attachment; filename="${filename}.tsv"`,
-                    );
-                    axios
-                        .get<string>(
-                            'https://raw.githubusercontent.com/globaldothealth/list/main/data-serving/scripts/export-data/functions/01-split/fields.txt',
-                        )
-                        .then((txtRes) => {
-                            const columns = txtRes.data.split('\n');
-                            const parsedCases = _.map(
-                                matchingCases,
-                                parseDownloadedCase,
-                            );
-                            const stringifiedCases = stringify(parsedCases, {
-                                header: true,
-                                columns: columns,
+                    await axios.get<string>(CaseFieldTextFile)
+                    .then((txtRes) => {
+                        columnsList = txtRes.data.split('\n');
+                        const columnsString = columnsList.join('\t');
+                        res.write(columnsString);
+                        res.write('\n');
+                    });
+
+                    try {
+                        let doc: CaseDocument;
+                        doc = await cursor.next();
+                        while (doc != null) {
+                            delete doc.restrictedNotes;
+                            const parsedCase = parseDownloadedCase(doc);
+                            const stringifiedCase = stringify([parsedCase], {
+                                header: false,
+                                columns: columnsList,
                                 delimiter: '\t',
                             });
-                            res.end(stringifiedCases);
-                        });
+                            res.write((stringifiedCase));
+                            doc = await cursor.next();
+                        }
+                    } catch(err) {
+                        logger.error(`Error streaming case data: ${err}`);
+                        res.status(500).json('A server error occurred while streaming case data');
+                        return;
+                    }
+                    res.end();
+                    
                     break;
 
                 case 'json':
```

Comment:
Looks good, maybe we should think about writing JSONL, since that's more natural for lists of records https://jsonlines.org/. Again, without user surveys, not sure which would be more useful. JSONL does make command line parsing easier (every record on its own line). Disadvantages: less well-known than JSON.

Reply:
Maybe. Gonna hold off for now and hoping command line users know `jq`.